### ISSUE TITLE
Overlay doesn't display after switching controllers

### ISSUE THREAD

**Issue by zkxs** [Curtis-VL/OVRToolkit-Issues#128]
## Summary
I have two pairs of index controllers, and when one set runs out of battery I rotate them out. When I do this, the wrist overlay no longer appears when I look at my wrist. The window overlay can still be opened with the new controllers, and works as expected: only the wrist overlay is affected.

## To reproduce
1. Enter SteamVR normally
2. Power on both hand controllers
3. Power off both hand controllers
4. Power on a different set of hand controllers

## Workaround
Close and reopen OVR Toolit

**Comment by Curtis-VL** [comment]
I'm afraid this is 'somewhat' intended behaviour at the moment.

The issue really stems from SteamVR labelling trackers as left/right hands unless you set the tracker roles as 'None' in the SteamVR settings, which most users haven't done. :)

This causes OVR Toolkit to become bound to a waist tracker for example, kinda messing up everything. As a result, right now it only assigns left/right controller roles on startup then never again throughout the session.

Potentially... I could check if there's multiple devices of the same controller type, such as multiple left knuckles controllers. (Even when disconnected, unless you restart SteamVR the old controller still exists) In this situation I could ask SteamVR for left/right controller roles again.

I'll have a look into this! I don't have a second set of controllers so it'll have to be all 'in theory' though! If you're in our Discord give me an @Curtis#1149 and I can have you test the changes once I've made some!

**Comment by nullstalgia** [comment]
I wonder if you could just ask for the roles each time a new device appears.

**Comment by zkxs** [comment]
Interesting. I do have Vive trackers, which I've gone through and manually tagged as left foot, right foot, hips, etc. I'm happy to help any way I can if you want me to test things. I'll ping you on Discord.

**Comment by nullstalgia** [comment]
We typically recommend leaving the roles to None due to some people's Wristwatches being attached to a tracker instead of the controller, just so you know.

**Comment by zkxs** [comment]
I've just tested again, and even with no Vive trackers connected swapping hand controllers still makes the wrist overlay go away. If I swap back to the original controllers, the overlay comes back.

I'm not sure why tagging a tracker as say, Left Foot, would cause it to be confused for a hand. Unless @nullstalgia is saying that merely having the Vive trackers tagged, even if disconnected, is enough to induce this issue? Maybe I'm misunderstanding what's going on with this tracker tagging problem.

**Comment by nullstalgia** [comment]
Sorry for the confusion ^^

What I was trying to convey was that we occasionally get support requests from people who have the watch attached to their foot or something, and so we usually tell them to blank out the tracker Roles. And so I was just sharing that as a preventative measure, not a solution for the current problem. :)

(But also, if Curtis is able to find a way to reliably determine the correct L/R hand controllers, then that issue might also be fixed!)

**Comment by Curtis-VL** [comment]
Hey @zkxs,

Could you confirm if the wrist overlay switches to the correct hand again when you toggle your windows on via double pressing both A buttons?

Having a look through the logic, it looks like OVR Toolkit checks if there's a mismatch between its own 'left controller index' versus the SteamVR 'controller role' for left controller, but this only happens when you toggle your overlays on/off to avoid overhead from constantly running this check.

**Comment by zkxs** [comment]
> Could you confirm if the wrist overlay switches to the correct hand again when you toggle your windows on via double pressing both A buttons?

@Curtis-VL I use double-grip to toggle my windows on and off. That's the same thing, right? If so, then toggling windows doesn't reset the wrist overlay to the new controller.

**Comment by Curtis-VL** [comment]
You're right, on anything but Index it's double Grip, on Index it's double A. (With double grip as a legacy binding)

That's interesting! On performing that binding it asks SteamVR for the left and right controller roles, maybe SteamVR isn't changing the controller role? Hmmm.

**Comment by zkxs** [comment]
Hmm, I can reproduce it again and capture one of those fancy SteamVR System Reports that Steam Support likes so much, if that'd help. Not sure if I want to attach a system report directly to the GitHub issue though, as those reports have *a lot* of detail in them and frankly I'm not sure how sensitive some of it is.
